pycparser/c_lexer.py

```diff
@@ -475,6 +479,16 @@ def t_BAD_CONST_OCT(self, t):
         msg = "Invalid octal constant"
         self._error(msg, t)
 
+    @TOKEN(unsupported_c89_comment)
+    def t_UNSUPPORTED_C89_COMMENT(self, t):
+        msg = "Comments are not supported (note: GCC's cpp removes comments, Clang's cpp does not)."
```

Comment:
This error message is way too specific; you have a particular issue you're encountering now, but there are a million other reasons people could try to run pycparser on files with comments. It could point to the repository's README, for example (which has a section on the preprocessor)

Reply:
Thanks for the context, I'll point to the README, but I'd also like to add more context to the README.

Is his diagram an accurate understanding of the situation?  
<img width="763" alt="Screenshot 2025-04-02 at 12 09 23 AM" src="https://github.com/user-attachments/assets/2af55359-7ca1-472f-9584-84d7df3753c3" />